chingsley/node-auth3-guided
# - integrate jwt for authorization

#### What does this PR do?
It implements authorization with `jsonwebtoken` (JWT).

#### Description of Task to be completed?
- Install npm package jsonwebtoken.
- Rewrite the authorization middlewares to use JWT.
- Implement the `checkRole` middleware to protect admin roles 

#### How should this be manually tested?
- *In postman*, log in with a POST request to the endpoint `http://localhost:3000/api/auth/login`, providing your `username` and `password` in the request body. This action generates a token.
- Try to access the endpoint`http://localhost:3000/api/users/:id` while providing the token in the header as the key `Authorization`. Confirm that you have such permission. Note: by default, users are assigned the 'student' role.
- Try to access the endpoint`http://localhost:3000/api/users` while providing the token in the header as the key `Authorization`. Confirm that you are denied the privilege. Such an endpoint has been reserved for admin roles only.

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
N/A

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A
